tufaceous-lib/src/oxide_metadata.rs

```diff
@@ -52,8 +49,9 @@ impl Metadata {
         let mut b = serde_json::to_vec(self)?;
         b.push(b'\n');
 
-        let mtime =
-            SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+        // XXX This was changed from upstream to add oxide.json with a zero
+        // timestamp, to ensure stability of fake manifests.
```

Comment:
Should this method take an `MtimeSource`?

Reply:
Hmm, I feel like I tried that and it didn't quite work, but I might be wrong -- let me check.

Reply:
Oh yeah, the thing I ran into is that at the moment we don't have mtime source annotations in a few spots. For example,

```toml
[[artifact.host]]
name = "fake-host"
version = "1.0.0"
[artifact.host.source]
kind = "composite-host"
phase_1 = { kind = "fake", size = "512KiB" }
phase_2 = { kind = "file", path = "foo/bar" }
```

In this case, what should the mtime of `oxide.json` be? I punted on that.

However, on a re-look I realized that we can just base this on whether all of the sub-artifacts are fake. Updated the PR with this new approach.

Thanks for pushing me on this!